src/core_ocean/shared/mpas_ocn_tracer_hmix_redi.F

```diff
@@ -143,6 +142,7 @@ subroutine ocn_tracer_hmix_Redi_tend(meshPool, scratchPool, layerThickness, laye
       real(kind=RKIND), dimension(:), pointer :: areaCell, dvEdge, dcEdge
       real(kind=RKIND), dimension(:), allocatable :: minimumVal
       real(kind=RKIND), dimension(:, :), allocatable :: fluxRediZTop
+      real(kind=RKIND), dimension(:,:,:), allocatable :: redi_term1, redi_term2, redi_term3
```

Comment:
@vanroekel for allocated arrays that have a shared dimension amongst threads (iCell, iEdge, iVertex), the declaration needs to be put at the top of the module. Now passes bfb thread test. I'll push up the change.